File: app/services/semantic_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class SemanticService {
  static const String _baseUrl = 'http://192.168.2.94:8000'; // Local backend

  static Future<List<Map<String, dynamic>>> getTagsFromText(String inputText) async {
    try {
      final url = Uri.parse('$_baseUrl/semantic/match');
      final response = await http.post(
        url,
        headers: {'Content-Type': 'application/json'},
        body: jsonEncode({'user_input': inputText}),
      );

      if (response.statusCode == 200) {
        final decoded = jsonDecode(response.body);
        final matches = decoded['matches'] as List<dynamic>;

        return matches.map((m) {
          return {
            'tag': m['tag'] ?? '',
            'score': (m['score'] ?? 0.1).toDouble(),
            'category': m['category'] ?? 'unknown',
          };
        }).toList();
      } else {
        print('❌ Semantic API failed with status: ${response.statusCode}');
      }
    } catch (e) {
      print('❌ Error calling semantic API: $e');
    }

    return [];
  }
}

File: app/services/semantic_service.py
import requests

SEMANTIC_API_URL = "http://localhost:8000/semantic/match"

def get_semantic_tags(user_input: str):
    try:
        print(f"📤 Sending to: {SEMANTIC_API_URL}")
        print(f"📨 Payload: {{'user_input': '{user_input}'}}")

        response = requests.post(
            SEMANTIC_API_URL,
            json={"user_input": user_input}
        )

        print(f"🔧 Status Code: {response.status_code}")
        print(f"📦 Raw Response: {response.text}")

        if response.status_code == 200:
            return response.json().get("matches", [])
        else:
            return []
    except Exception as e:
        print(f"❌ Error calling semantic API: {e}")
        return []

